src/Corax/Utils/SpatialHelper.cs

```diff
@@ -0,0 +1,280 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Runtime.CompilerServices;
+using Sparrow.Server;
+using Spatial4n.Core.Context;
+using Spatial4n.Core.Distance;
+using Spatial4n.Core.Shapes;
+using Spatial4n.Core.Util;
+using Voron;
+using Voron.Data.CompactTrees;
+
+namespace Corax.Utils;
+
+public class SpatialHelper
+{
+    //Source: https://github.com/apache/lucenenet/blob/master/src/Lucene.Net.Spatial/Query/SpatialArgs.cs
+    private static readonly char[] alphabet =
+    {
+        '0', '1', '2', '3', '4', '5', '6', '7', '8', '9', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'j', 'k', 'm', 'n', 'p', 'q', 'r', 's', 't', 'u', 'v', 'w', 'x', 'y', 'z'
+    };
+
+    public const int DefaultGeohashLevel = 9;
+    private static readonly Dictionary<string, IRectangle> CachedFigures;
+    private const int CachePrefixLength = 4;
+    private const int Threshold = 2 << 10;
+    
+    public enum SpatialUnits
```

Comment:
Please create `Corax/Utils/Spatials` directory and move the classes / enums as separate files there